sdk/servicebus/azure-servicebus/azure/servicebus/aio/_async_utils.py

```diff
@@ -7,14 +7,48 @@
 import asyncio
 import logging
 import datetime
+import functools
 
-from azure.servicebus.common.utils import renewable_start_time, get_running_loop
-from azure.servicebus.common.errors import AutoLockRenewTimeout, AutoLockRenewFailed
+from uamqp import authentication
+
+from .._common.utils import renewable_start_time, get_running_loop, utc_now
+from ..exceptions import AutoLockRenewTimeout, AutoLockRenewFailed
+from .._common.constants import JWT_TOKEN_SCOPE
 
 
 _log = logging.getLogger(__name__)
 
 
+async def create_authentication(client):
+    # pylint: disable=protected-access
+    try:
+        # ignore mypy's warning because token_type is Optional
+        token_type = client._credential.token_type  # type: ignore
+    except AttributeError:
+        token_type = b"jwt"
+    if token_type == b"servicebus.windows.net:sastoken":
```

Comment:
If used in multiple places may be worth constifying; if only used here maybe locally to not clutter the shared file.  (If only used in one location ever feel free to use your judgement)

Reply:
updated to constants!